libsolidity/ast/Types.cpp

```diff
@@ -1661,6 +1662,37 @@ MemberList::MemberMap ContractType::nativeMembers(ContractDefinition const*) con
 				&it.second->declaration()
 			));
 	}
+	// Add overloads from address only if there is no conflict
+	for (auto it = addressMembers.begin(); it != addressMembers.end(); ++it)
+	{
+		bool clash = false;
+		for (auto const member: members)
+		{
+			if (
+				member.name == it->name &&
+				(
+					// Members with different types are not allowed
+					member.type->category() != it->type->category() ||
+					// Members must overload functions without clash
+					(
+						member.type->category() == Type::Category::Function &&
+						dynamic_cast<FunctionType const&>(*member.type).hasEqualArgumentTypes(dynamic_cast<FunctionType const&>(*it->type))
```

Comment:
This is needed because simply comparing the type takes visibility specifiers into account, which is a can of worms.